### PR TITLE
[WIP] Account abstraction with tx bundler

### PR DESCRIPTION
**Pull request by jmederosalvarado**
## Changes:
- Introduces TxBundler approach in AccountAbstraction plugin

## Types of changes

What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.AccountAbstraction/Bundler/TxBundleRegularlyTrigger.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+{
+    public class TxBundleRegularlyTrigger : ITxBundlingTrigger
+    {
+        private readonly Timer _timer;
+
+        public TxBundleRegularlyTrigger() : this(TimeSpan.FromSeconds(2)) { }
+
+        public TxBundleRegularlyTrigger(TimeSpan interval)
+        {
+            _timer = new Timer(interval.TotalMilliseconds);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.AccountAbstraction/Bundler/TxBundleRegularlyTrigger.cs:14]: Maybe PeriodicTimer will be better (dunno the requirments)

File: src/Nethermind/Nethermind.AccountAbstraction/Bundler/TxBundler.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            _trigger.TriggerTxBundling -= OnTriggerTxBundling;
+            _bundlerCancellationToken?.Dispose();
+            return Task.CompletedTask;
+        }
+
+        private void OnTriggerTxBundling(object? sender, EventArgs e)
+        {
+            BlockHeader head = _blockTree!.Head!.Header;
+            Transaction? tx = _source.GetTransaction(head, _gasLimitProvider.GetGasLimit());
+            if (tx is null) return;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.AccountAbstraction/Bundler/TxBundler.cs:56]: Minor: I would prefer 
if (tx is not null)

File: src/Nethermind/Nethermind.AccountAbstraction/Bundler/TxBundlerFlashbots.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            return Task.CompletedTask;
+        }
+
+        private void OnTriggerTxBundling(object? sender, EventArgs e)
+        {
+            BlockHeader head = _blockTree.Head!.Header;
+            Transaction? tx = _source.GetTransaction(head, _gasLimitProvider.GetGasLimit());
+            if (tx is null) return;
+
+            FlashbotsSender flashbotsSender = new(new HttpClient(), _signer, _logger);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.AccountAbstraction/Bundler/TxBundlerFlashbots.cs:67]: https://softwareengineering.stackexchange.com/questions/330364/should-we-create-a-new-single-instance-of-httpclient-for-all-requests

File: src/Nethermind/Nethermind.AccountAbstraction/Bundler/TxBundlerFlashbots.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+            FlashbotsSender.MevBundle bundle =
+                new(_blockTree.Head.Header.Number + 1, new[] { Rlp.Encode(tx).ToString() });
+            flashbotsSender.SendBundle(bundle, _flashbotsEndpoint).ContinueWith(_ => _);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.AccountAbstraction/Bundler/TxBundlerFlashbots.cs:71]: ).ContinueWith(_ => _); ???

File: src/Nethermind/Nethermind.AccountAbstraction/Source/UserOperationTxSource.cs [hunk 5]
```diff
@@ -49,17 +48,17 @@ public class UserOperationTxSource : ITxSource
...
             IList<UserOperation> userOperationsToInclude = new List<UserOperation>();
             ulong gasUsed = 0;
 
-            IEnumerable<UserOperation> userOperations = 
+            IEnumerable<UserOperation> userOperations =
                 _userOperationPool
                     .GetUserOperations()
-                    .Where(op => op.MaxFeePerGas >= parent.BaseFeePerGas)
-                    .OrderByDescending(op => CalculateUserOperationPremiumGasPrice(op, parent.BaseFeePerGas));
+                    .Where(op => op.MaxFeePerGas >= head.BaseFeePerGas)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.AccountAbstraction/Source/UserOperationTxSource.cs:60]: Not sure what GetUserOperations returns but this is potential place for optimization

**Comment by jmederosalvarado** [comment]
We decided to go a different path. So I'm closing this. Thanks for the feedback @LukaszRozmej.

**Pull request closed without merging**